# Problem Set 5
### Written by Adithya Solai

_Copyright © 2021 Adithya Solai. All rights are reserved.
You cannot use, modify, or redistribute this code without 
explicit permission from Adithya Solai._

In [1]:
import numpy as np
import pandas as pd

In this problem set, you will be exclusively working with data on MLB (Major League Baseball) teams from 1990 to 2014. You should have been given a baseball.csv file with the data in it (look in the `/datasets` folder). Make sure that .csv file is in the same file directory as this .ipynb file. Run the code in the cell below to load the data from this csv into a pandas DataFrame.

In [3]:
baseball_df = pd.read_csv("baseball.csv")
baseball_df.head()

teamID  yearID  total_payroll franchID   W    G
0    BAL    1990        9680084      BAL  76  161
1    BOS    1990       20558333      BOS  88  162
2    CAL    1990       21720000      ANA  80  162
3    CHA    1990        9491500      CHW  94  162
4    CLE    1990       14487000      CLE  77  162

As seen above, this dataset tracks the # of wins each team got in a season (`W` column), the total # of games in that season (`G` column), and the amount of money in $ that the team spent on their player salaries that year (`total_payroll` column). This is done for all years from 1990 to 2014 (`yearID` column)

## Complete all tasks below.

### Task 1
**Drop the `teamID` column. The teamID changes throughout the years, as teams re-brand themselves with different cities/mascots/acronyms. The `franchId` (franchise ID) stays consistent throughout the years, so that column will be most helpful to our analysis.**

### Task 2
**Find the minimum and maximum value in the `yearID` column to ensure that this dataset is actually from 1990 to 2014.**

### Task 3
**Create a new column called `win_pct` that represents the win percentage of each team for a particular season. Use the `W` and `G` column to easily create the new column. The `win_pct` column should be some `float` datatype, and all values in this column should be between 0 and 1 since it is a percentage.**

For example, the `win_pct` column for the first row in the dataset (`franchID`=BAL, `yearID`=1990) would be 76 / 161, or 0.47205.

### Task 4
**Print the mean payroll for the New York Yankees (`franchID`=NYY) in the years 2000-2014. Do the same for the Oakland Athletics (`franchID`=OAK).**

### Task 5
**Find the team with the lowest average payroll from 1990-2015 (the entire dataset). Find the team with the highest average payroll from 1990-2014 as well.**

### Task 6
**Create a new column called `payroll_efficiency`, which is basically the `total_payroll` column divided by the `W` column. This will give a rough estimate of how efficient each team is at using their payroll to get wins.**

A lower `payroll_efficiency` value means that the team is more efficient since a low value is the result of a low numerator (low payroll spending) and a high denominator (high # of wins).

### Task 7
**Find the team with the lowest average `payroll_efficiency` from 1990-2014 to get a rough estimate of the most efficient team. Find the team with the highest average `payroll_efficiency` to find the team that is least efficient.**